pkg/doc/verifiable/linked_data_proof.go

```diff
@@ -0,0 +1,130 @@
+package verifiable
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/signer"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+)
+
+// signatureSuite encapsulates signature suite methods required for signing documents
+type signatureSuite interface {
+
+	// GetCanonicalDocument will return normalized/canonical version of the document
+	GetCanonicalDocument(doc map[string]interface{}) ([]byte, error)
+
+	// GetDigest returns document digest
+	GetDigest(doc []byte) []byte
+
+	// Accept registers this signature suite with the given signature type
+	Accept(signatureType string) bool
+}
+
+type verifierSignatureSuite interface {
+	signatureSuite
+
+	// Verify will verify signature against public key
+	Verify(pubKey []byte, doc []byte, signature []byte) error
+}
+
+type signerSignatureSuite interface {
+	signatureSuite
+
+	// Sign  will sign JSON LD document
+	// todo refactor, do not pass privateKey (https://github.com/hyperledger/aries-framework-go/issues/339)
+	Sign(jsonLdDoc, privKey []byte) ([]byte, error)
+}
+
+type keyResolverAdapter struct {
+	pubKeyFetcher PublicKeyFetcher
+}
+
+func (k *keyResolverAdapter) Resolve(id string) ([]byte, error) {
+	fetcher, err := k.pubKeyFetcher("", id)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeyBytes, ok := fetcher.([]byte)
+	if !ok {
+		return nil, errors.New("expecting []byte public key, got something else")
+	}
+
+	return pubKeyBytes, nil
+}
+
+// LinkedDataProofContext holds options needed to build a Linked Data Proof.
+// todo refactor, do not pass privateKey (https://github.com/hyperledger/aries-framework-go/issues/339)
+type LinkedDataProofContext struct {
+	SignatureType string               // required
+	Suite         signerSignatureSuite // required
+	PrivateKey    []byte               // required
```

Comment:
This should be addressed soon

Reply:
yes - we need these crypto operations & key holding to handled by Crypto/KMS.